internal/objectstore/base.go

```diff
@@ -53,15 +53,16 @@ type opType int
 
 const (
 	opGet opType = iota
-	opGetByHash
+	opGetBySHA256
+	opGetBySHA256Prefix
 	opPut
 	opRemove
 )
 
 type request struct {
 	op            opType
 	path          string
-	sha256Prefix  string
+	sha256        string
```

Comment:
I'm not entirely happy with how this is overloaded. To me, this points that we should do something different here.

Perhaps an interface instead:
```
type request interface {
   opType() opType
}
```

Then, based on a switch of `opType`, cast to various implementations?
```
type getRequest struct {
   path string
}

func (getRequest) opType() opType {
  return opGet
}
```

I don't think this needs to be changed now though.

What do you think?

Reply:
I agree. I'll sweep back around to this, as I need to add more to this in another PR.